File: app/src/main/java/com/grsu/guideapp/network/APIService.java
package com.grsu.guideapp.network;

import com.grsu.guideapp.network.model.Datum;
import com.grsu.guideapp.network.model.Root;
import java.util.List;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface APIService {

    @GET("mobile/routes/category/101?Lang=all")
    Call<Root> getRoutes(@Query("apikey") String apikey);

//    @GET("object/{id}?Lang=all")
    @GET("mobile/route/{id}?Lang=all")
    Call<Datum> getPoi(@Path("id") long id, @Query("apikey") String apikey);

    @GET("mobile/route/{id}?Lang=all")
    Call<Datum> getRouteById(@Path("id") long id, @Query("apikey") String apikey);

    @GET("external/search/objects?Lang=all&category=101")
    Call<Root> checkUpdateRoute(
            @Query("apikey") String apikey,
            @Query("update_datetime") String update_datetime
    );

    @GET("external/search/objects?Lang=all")
    Call<Root> checkUpdatePoi(
            @Query("apikey") String apikey,
            @Query("update_datetime") String update_datetime
    );

    @GET("mobile/route/{id}/objects?Lang=all")
    Call<List<Datum>> updatePoi(@Path("id") long id, @Query("apikey") String apikey);
}

File: app/src/main/java/com/grsu/guideapp/network/APIService.kt
package com.grsu.guideapp.network

import com.grsu.guideapp.network.model.Datum
import com.grsu.guideapp.network.model.Root
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

@Suppress("unused")
interface APIService {

    @GET("mobile/routes/category/101?Lang=all")
    fun getRoutes(): Call<Root>

    @GET("mobile/route/{id}?Lang=all")
    fun getPoi(@Path("id") id: Long): Call<Datum>

    @GET("mobile/route/{id}?Lang=all")
    fun getRouteById(@Path("id") id: Long): Call<Datum>

    @GET("external/search/objects?Lang=all&category=101")
    fun checkUpdateRoute(@Query("update_datetime") update_datetime: String): Call<Root>

    @GET("external/search/objects?Lang=all")
    fun checkUpdatePoi(@Query("update_datetime") update_datetime: String): Call<Root>

    @GET("mobile/route/{id}/objects?Lang=all")
    fun updatePoi(@Path("id") id: Long): Call<List<Datum>>
}